docs/dev-proc/source-code-structure.md

```diff
@@ -84,29 +123,29 @@ The procedure of merging is as follows:
 1. Make sure to receive at least one `Approve` in the review process.
 1. Make sure that **all** change requests are resolved.
 1. Merge the branch using git CLI. In case of merging the `feature` branch into
-   `develop` branch it may look as follows:
+   `dasharo` branch it may look as follows:
 
     ```bash
     git fetch dasharo
-    git checkout dasharo/<platform>/develop -b <platform>/develop
-    git merge --ff-only dasharo/<platform>/<feature>
-    git push dasharo <platform>/develop
+    git checkout origin/dasharo -b dasharo
+    git merge --ff-only origin/<feature>
+    git push origin dasharo
     ```
 
-1. This should automatically trigger closing the MR in the GitHub web UI.
-1. The remote branch can be safely deleted after this process.
-
-    ```bash
-    git push dasharo --delete <platform>/<feature>
-    ```
+1. This should automatically trigger closing the MR and deleting the merged
+   branch on GitHub.
 
 1. Note that the merging may fail if the source (in this case: `feature`) branch
-   is not properly rebased on top of the target (in this case: `develop`)
+   is not properly rebased on top of the target (in this case: `dasharo`)
    branch. In such a case, one must rebase the source branch first:
 
    ```bash
-   git checkout dasharo/<platform>/<feature>
-   git checkout -b <platform>/<feature>
-   git rebase dasharo/<platform>/<develop>
-   git push -f dasharo <platform>/<feature>
+   git checkout origin/<feature>
+   git checkout -b <feature>
+   git rebase origin/dasharo
+   git push -f origin <feature>
    ```
+
+> Remember to push the rebased branch _before_ merging it to `dasharo`.
```

Comment:
In rare cases something may break after rebase. it would be good to at least wait for CI check or retest given feature

Reply:
added a comment in 42ea12ac580f49fa04e7a16cdd86f633e468426a